Refactor the DataGrid and PivotGrid export snippets (#1235)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/dxDataGrid/1 Configuration/export/export.md

The following instructions show how to enable and configure client-side export:

1. **Install or reference the required libraries**  

>     You need the following libraries: <a href="https://github.com/exceljs/exceljs" target="_blank">ExcelJS</a> v4.1.1 and <a href="https://github.com/eligrey/FileSaver.js/" target="_blank">FileSaver</a> v2.0.2.

Reviewer: ```suggestion
    This feature requires <a href="https://github.com/exceljs/exceljs" target="_blank">ExcelJS</a> v4.1.1+ and <a href="https://github.com/eligrey/FileSaver.js/" target="_blank">FileSaver</a> v2.0.2+.
```

    ---
    ##### jQuery
        <!--HTML-->
        <head>
            <!-- ... -->
            <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/7.4.0/polyfill.min.js"></script>
            <script src="https://cdnjs.cloudflare.com/ajax/libs/exceljs/4.1.1/exceljs.min.js"></script>
            <script src="https://cdnjs.cloudflare.com/ajax/libs/FileSaver.js/2.0.2/FileSaver.min.js"></script>
            <!-- reference the DevExtreme sources here -->
        </head>

    ##### Angular   

>         <!-- tab: Installation command -->

Reviewer: If you remove the tab name, will it be displayed one under another with tsconfig.app.json?

Author: I'm not really sure that I got you right, but different options (`<!-- -->`, `<!-- tab: -->` or without it at all) produce the following output:
![image](https://user-images.githubusercontent.com/33040934/93361509-9db0c780-f84d-11ea-9958-44196567868d.png)
![image](https://user-images.githubusercontent.com/33040934/93361543-a6090280-f84d-11ea-8276-8bd61c8e160d.png)

Reviewer: Okay, Let's keep it as is.

>         npm install --save exceljs@4.1.1 file-saver

Reviewer: Doesn't the command without specified version install the latest version?

        <!-- tab: tsconfig.app.json -->
        "compilerOptions": {
            // ...
            "paths": {
                // ...
                "exceljs": [
                    "node_modules/exceljs/dist/exceljs.js"
                ]
            }
        }

>         <!-- tab: app.component.ts -->

Reviewer: Nothing happens here. We can remove this tab.

The following instructions show how to enable and configure client-side export:

1. **Install or reference the required libraries**  

>     This feature requires <a href="https://github.com/exceljs/exceljs" target="_blank">ExcelJS</a> v3.3.1+ and <a href="https://github.com/eligrey/FileSaver.js/" target="_blank">FileSaver</a> v2.0.2+ for all frameworks except Angular. The latter requires `ExcelJS` v1.7.0.

Reviewer: How about we make it a list like here?
https://github.com/DevExpress/devextreme-documentation/pull/1251/files#diff-d384da97e40226a28ff7c62338331f51R5

The following instructions show how to enable and configure client-side export:

1. **Install or reference the required libraries**  

>     This feature requires <a href="https://github.com/exceljs/exceljs" target="_blank">ExcelJS</a> v4.1.1+ and <a href="https://github.com/eligrey/FileSaver.js/" target="_blank">FileSaver</a> v2.0.2+.

Reviewer: Looks like this should be changed in the wake of today's discussion on Teams.

The following instructions show how to enable and configure client-side export:

> 1. **Reference the required libraries**  

Reviewer: ```suggestion
1. **Install or reference the required libraries**  
```

>     Reference or import the <a href="https://github.com/exceljs/exceljs" target="_blank">ExcelJS</a> and <a href="https://github.com/eligrey/FileSaver.js/" target="_blank">FileSaver</a> libraries

Reviewer: The imperative verbs "reference" and "install" are already present in the item title. In this paragraph, it would be better to specify which versions are supported.

    ---
    ##### jQuery
        <!--HTML-->
        <head>
            <!-- ... -->
            <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/7.4.0/polyfill.min.js"></script>
            <script src="https://cdnjs.cloudflare.com/ajax/libs/exceljs/4.1.1/exceljs.min.js"></script>
            <script src="https://cdnjs.cloudflare.com/ajax/libs/FileSaver.js/2.0.2/FileSaver.min.js"></script>
            <!-- reference the DevExtreme sources here -->
        </head>

    ##### Angular   

        <!-- tab: Installation command -->
        npm install --save exceljs@4.1.1 file-saver

        <!-- tab: tsconfig.app.json -->

>         // add ExcelJS to the *"paths"* section:

Reviewer: This is kinda obvious, don't you think?

        "compilerOptions": {
            // ...
            "paths": {
                // ...
                "exceljs": [
                    "node_modules/exceljs/dist/exceljs.js"
                ]
            }
        }

        <!-- tab: app.component.ts -->
        import { Component } from '@angular/core';
        import { Workbook } from 'exceljs';

>         import saveAs from 'file-saver';

Reviewer: I noticed that neither ExcelJS nor FileSaver are used until p.3. Maybe we should wait with showing the imports until that time.

The following instructions show how to enable and configure client-side export:

1. **Reference the required libraries**  

>     Reference or import the <a href="https://github.com/exceljs/exceljs" target="_blank">ExcelJS</a> v3.3.1 or newer and <a href="https://github.com/eligrey/FileSaver.js/" target="_blank">FileSaver</a> libraries. If you use Angular, import **ExcelJS** v1.7.0 and add this library to the `tsconfig.app.json` file.

Reviewer: Not only import but you also need to install the libraries. I suggest splitting the entire instructions by frameworks:

---
##### jQuery

Reference the libraries:

... code ...

##### Angular

Install the libraries:

> npm install ...

Import them:

... code ...

##### Vue 

... and so on ...

        })
        export class AppModule { }

    ##### Vue

>         <!-- tab: Installation command -->

Reviewer: ```suggestion

```

        npm install --save exceljs@4.1.1 file-saver

>         <!-- tab: App.vue -->

Reviewer: And this.

    ##### React

>         <!-- tab: Installation command -->

Reviewer: ```suggestion

```

        npm install --save exceljs@4.1.1 file-saver

>         <!-- tab: App.js -->

Reviewer: And this.

>         <!-- tab: app.component.ts -->

Reviewer: Nothing happens here, too.

    ---

3. **Export the DataGrid**   

>     Implement the [onExporting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onExporting) handler and call the [exportDataGrid(options)](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) method in it. In the code below, this method exports the **DataGrid** as is, but you can use [ExportDataGridProps](/Documentation/ApiReference/Common/Object_Structures/ExportDataGridProps/) to configure export settings, including [cell customization](/Documentation/ApiReference/Common/Object_Structures/ExportDataGridProps/#customizeCell). The **DataGrid** is exported to an Excel worksheet that is created using the ExcelJS API. To save the Excel document, call the FileSaver's **saveAs** method. The `e.cancel` parameter disables the deprecated built-in export implementation with fewer capabilities.

Reviewer 2: "The `e.cancel` parameter disables the deprecated built-in export implementation with fewer capabilities."
\+ It is required until we completely remove the deprecated implementation. - Do we need this clarification here (+ pivot)?

Author: I think we do, we show this in code and should clarify this odd-looking line. When we'll manage to remove the old export I'll also remove this line from docs.

    ---

3. **Export the DataGrid**   

>     Import the `ExcelJS` and `FileSaver` libraries. Then implement the [onExporting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onExporting) handler and call the [exportDataGrid(options)](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) method in it. In the code below, this method exports the **DataGrid** as is, but you can use [ExportDataGridProps](/Documentation/ApiReference/Common/Object_Structures/ExportDataGridProps/) to configure export settings, including [cell customization](/Documentation/ApiReference/Common/Object_Structures/ExportDataGridProps/#customizeCell). To save the Excel document, call the FileSaver's **saveAs** method. The `e.cancel` parameter disables the deprecated built-in export implementation with fewer capabilities.

Reviewer: jQuery doesn't import the libraries. We need to rephrase the paragraph without explicitly saying "import".

Author: Maybe begin the paragraph as follows:
"This feature requires that you import the `ExcelJS` and `FileSaver` libraries in the modular environment. Reference these libraries if you use jQuery. Then implement..."?

Reviewer: We do the referencing in step 1. I think we should mention the libraries where we use them, something like:

```suggestion
    Implement the [onExporting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#onExporting) handler and call the [exportDataGrid(options)](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) method in it. In the code below, this method exports the **DataGrid** as is, but you can use [ExportDataGridProps](/Documentation/ApiReference/Common/Object_Structures/ExportDataGridProps/) to configure export settings, including [cell customization](/Documentation/ApiReference/Common/Object_Structures/ExportDataGridProps/#customizeCell). The **DataGrid** is exported to an Excel worksheet that is created using the ExcelJS API. To save the Excel document, call the FileSaver's **saveAs** method. The `e.cancel` parameter disables the deprecated built-in export implementation with fewer capabilities.
```

                        excelCell.alignment = { horizontal: 'left' };
                    } 
                }).then(function() {

>                     workbook.xlsx.writeBuffer().then(function(buffer: BlobPart) { 

Reviewer: ```suggestion
                    workbook.xlsx.writeBuffer().then(function(buffer) { 
```

        })
        export class AppComponent {
            onExporting(e) {

>                 const workbook = new Workbook();    

Reviewer: Line 329: Angular won't compile until I specify the type for `buffer` as `BlobPart`

> workbook.xlsx.writeBuffer().then(function(buffer: BlobPart) {

Reviewer: This one is still waiting to be fixed.

                        } 
                    }).then(function() {
                        workbook.xlsx.writeBuffer()

>                             .then(function(buffer: BlobPart) {

Reviewer: ```suggestion
                            .then(function(buffer) {
```

                    } 
                }).then(function() {
                    workbook.xlsx.writeBuffer()

>                         .then(function(buffer: BlobPart) {

Reviewer: ```suggestion
                        .then(function(buffer) {
```
